sdk/core/AzureCore/Source/Pipeline/PipelineClient.swift

```diff
@@ -32,6 +32,27 @@ public protocol AzureConfigurable {
     var apiVersion: String { get }
     /// The `ClientLogger` to be used by the service client.
     var logger: ClientLogger { get }
+    /// Options for configuring telemetry sent by the service client.
+    var telemetryOptions: TelemetryOptions { get }
+}
+
+/// Options for configuring telemetry sent by the service client.
+public struct TelemetryOptions {
+    /// Whether platform information will be omitted from the user agent string sent by the service client.
+    public let telemetryDisabled: Bool
+    /// An optional user-specified application ID included in the user agent string sent by the service client.
+    public let applicationId: String?
```

Comment:
Can this be defaulted to the bundle identifier?

Reply:
We do this in the UserAgentPolicy, L77.

Reply:
I guess we could do here, but it seems better to do it in the policy since that avoids adding business logic to the options structure, and the policy has the various providers injected to its initializer already in order to decouple it from the operating system.

Reply:
Sounds good to me.